### ISSUE TITLE
Abstract readonly (and maybe other property modifiers) not enforced

### ISSUE THREAD

**Issue by Roam-Cooper** [Microsoft/TypeScript#14778]
**TypeScript Version:**  2.1.5

**Code**

```ts
abstract class A {
    abstract readonly b: any;
}

class C extends A {
    b: any;
}
```

**Expected behavior:**
Compiler should complain that Class C does not implement Property b as readonly.

**Actual behavior:**
Compiler accepts an implementation of Property b without the appropriate property modifier(s).

**Comment by RyanCavanaugh** [comment]
There's nothing wrong with `C`'s `b` being readwrite. It is still a correct subtype of `A`

**Comment by Roam-Cooper** [comment]
Is it possible for an abstract base class to force a subclass to implement a readonly property?

**Comment by RyanCavanaugh** [comment]
No. Derived types are always allowed to have "more capability" than their base types.
